src/engine/pieces/piece.ts

```diff
@@ -4,17 +4,31 @@ import Square from '../square';
 
 export default class Piece {
     public player: Player;
-
+    public firstMove;
     public constructor(player: Player) {
         this.player = player;
+        this.firstMove = true;
     }
 
     public getAvailableMoves(board: Board) {
         throw new Error('This method must be implemented, and return a list of available moves');
     }
 
+    public getLateralMoves(board: Board) {
```

Comment:
What's the reason behind the abstraction to lateral moves here? I don't think this is general-enough behaviour to belong to the `Piece` class for the time being, as at the moment it's specific to rooks.